presto-main/src/main/java/com/facebook/presto/operator/JoinStatisticsCounter.java

```diff
@@ -0,0 +1,65 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.operator;
+
+import com.facebook.presto.operator.LookupJoinOperators.JoinType;
+
+import java.util.function.Supplier;
+
+import static com.facebook.presto.operator.JoinOperatorInfo.createJoinOperatorInfo;
+import static java.util.Objects.requireNonNull;
+
+public class JoinStatisticsCounter
+        implements Supplier<OperatorInfo>
+{
+    public static final int HISTOGRAM_SIZE = 8;
```

Comment:
It is used in JoinOperatorInfo a lot and shared. I don't understand this "inline" thing. Rule "one usage => inline" doesn't increase readability. Name can comment and explain stuff. For example: `new int[100]` vs - `new int[MAX_CAPACITY]` vs `int[INITIAL_CAPACITY]`. Intention is clear right here. Having a named constant never hurt, it's a judgement call here and I think this make code more documented here.